### PR TITLE
Migrate Import and Export to redux

### PR DESCRIPTION
**Pull request by vilsbole**
[[ch7461]](https://app.clubhouse.io/mycrypto/story/7461/redux-step1-migrate-import-and-export)

### Completed
- Use redux store for import and exports
- Add e2e tests for both
- Prepare removal of legacy.reducer by deleting legacy action creators.
- Remove old `createActions` mocks. ie. This represents > 45 test files 😨 
- Fix ScreenUnlock. ie. It wrongfully relied on the return value of an action, so minor changes to adapt to event programming model. Full refactor and tests will be made with subsequent PR for `vault.slice`.

### Steps to test
- Can Export
- Can Import
- Can ScreenLock and ScreenUnlock

### CODE REVIEW

File: __tests__/export.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { FIXTURE_LOCALSTORAGE_WITH_ONE_ACC, FIXTURES_CONST, PAGES } from './fixtures';
+import SettingsPage from './settings-page.po';
+import { getTransValueByKey } from './translation-utils';
+
+function getFilePath(fileName) {
+  return join(homedir(), 'Downloads', fileName);
+}
+
+async function waitForFile(path) {
+  while (!(await existsSync(path))) await setTimeout(() => {}, FIXTURES_CONST.TIMEOUT);
```
Comment (GuillaumeRx) [inline __tests__/export.test.js:19]: Why awaiting on the `existsSync` synchronous call ?

File: __tests__/import.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const btn = getByText(getTransValueByKey('SETTINGS_IMPORT_COMPLETE'));
+  await t
+    .setFilesToUpload(getByTestId('upload-input'), ['./fixtures/persisted.json'])
+    .expect(btn)
+    .ok()
+    .click(btn)
+    .expect(getCurrentLocationUrl())
+    .eql(PAGES.SETTINGS, { timeout: FIXTURES_CONST.TIMEOUT })
+    // The file sets the Ethereum selectedNode to a custom node named "Emmanuel Macron"
+    .expect(getByText(/Emmanuel Macron/))
```
Comment (GuillaumeRx) [inline __tests__/import.test.js:33]: 🇫🇷🥖

File: src/features/ScreenLock/ScreenLockProvider.tsx [hunk 3]
```diff
@@ -84,57 +87,54 @@ class ScreenLockProvider extends Component<
...
 
   // Wipes encrypted data and unlocks
   public resetEncrypted = async () => {
-    const { destroyEncryptedCache } = this.props;
+    const { history, destroyEncryptedCache } = this.props;
     destroyEncryptedCache();
-    this.setState({ locked: false });
+    this.setState({ locked: false, decryptSuccess: false }, () => {
+      this.resetInactivityTimer();
+      history.replace(ROUTE_PATHS.DASHBOARD.path);
```
Comment (GuillaumeRx) [inline src/features/ScreenLock/ScreenLockProvider.tsx:129]: why not using `history.push` ?

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

File: __tests__/import.test.js (hunk 2 above)
Comment (FrederikBolding) [reply __tests__/import.test.js:33]: 😆

File: __tests__/export.test.js (hunk 1 above)
Comment (vilsbole) [reply __tests__/export.test.js:19]: Indeed! Removed.

File: src/features/ScreenLock/ScreenLockProvider.tsx (hunk 3 above)
Comment (vilsbole) [reply src/features/ScreenLock/ScreenLockProvider.tsx:129]: Initially I was surprised, but in fact from the moment the screen is unlocked you should not be able to return to a locked state with the router so `.replace` seems adequate.

Comment (GuillaumeRx) [reply src/features/ScreenLock/ScreenLockProvider.tsx:129]: That's a good catch ! 👍

**Review by GuillaumeRx — APPROVED** [review]
LGTM ! 🎉

**Pull request merged**